src/TokenHandler/AccountManager.mo

```diff
@@ -344,6 +354,24 @@ module {
       };
     };
 
+    /// Initiates a withdrawal by transferring tokens to another account.
+    /// Returns ICRC1 transaction index and amount of transferred tokens (fee excluded).
+    /// At the same time, it reduces the user's credit. Accordingly, amount < credit should be satisfied.
+    public func withdrawFromCredit(p : Principal, to : ICRC1.Account, amount : Nat) : async* WithdrawResponse {
+      if (amount > getCredit(p)) {
+        let err = #TooLowQuantity;
+        log(ownPrincipal, #withdrawalError(err));
+        return #err(err);
+      };
+      let result = await* withdraw(to, amount);
+      switch (result) {
+        // sync credit after successful withdrawal
+        case (#ok(_, _)) { debit_(p, amount) };
+        case (_) {};
+      };
+      result;
+    };
```

Comment:
Isn't it better to implement this in lib.mo instead of in AccountManager.mo? Then we don't need to pass `getCredit` to the AccountManager. AccountManager can remain unchanged.